docs/admin/templates/managing-templates/devcontainers/add-devcontainer.md

```diff
@@ -0,0 +1,143 @@
+   select a dev container-compatible template.
+
+1. Copy the files to your local device, then edit them to fit your needs.
+
+1. Upload them to Coder through the CLI or dashboard:
+
+   - CLI:
+
+   ```shell
+   coder templates push <template-name> -d <path to folder containing main.tf>
+   ```
+
+   - Dashboard:
+
+   1. Create a `.zip` of the template files:
+
+      - On Mac or Windows, highlight the files and then right click. A
+        "compress" option is available through the right-click context menu.
+
+      - To zip the files through the command line:
+
+        ```shell
+        zip templates.zip Dockerfile main.tf
+        ```
+
+   1. Select **Templates**.
+   1. Select **Create Template**, then **Upload template**:
+
+      ![Upload template](../../../../images/templates/upload-create-your-first-template.png)
+
+   1. Drag the `.zip` file into the **Upload template** section and fill out the
+      details, then select **Create template**.
+
+      ![Upload the template files](../../../../images/templates/upload-create-template-form.png)
+
+</div>
+
+To set variables such as the namespace, go to the template in your Coder
+dashboard and select **Settings** from the **⋮** (vertical ellipsis) menu:
+
+<Image height="255px" src="../../../../images/templates/template-menu-settings.png" alt="Choose Settings from the template's menu" align="center" />
+
+## Envbuilder Terraform provider
+
+When using the
+[Envbuilder Terraform provider](https://github.com/coder/terraform-provider-envbuilder),
+a previously built and cached image can be reused directly, allowing dev
+containers to start instantaneously.
+
+Developers can edit the `devcontainer.json` in their workspace to customize
+their development environments:
+
+```json
+…
+{
+  "features": {
+      "ghcr.io/devcontainers/features/common-utils:2": {}
+  }
+}
+…
+```
+
+## Example templates
+
+- [Docker dev containers](https://github.com/coder/coder/tree/main/examples/templates/devcontainer-docker)
+  - Docker provisions a development container.
+- [Kubernetes dev containers](https://github.com/coder/coder/tree/main/examples/templates/devcontainer-kubernetes)
+  - Provisions a development container on the Kubernetes cluster.
+- [Google Compute Engine dev container](https://github.com/coder/coder/tree/main/examples/templates/gcp-devcontainer)
+  - Runs a development container inside a single GCP instance. It also mounts
+    the Docker socket from the VM inside the container to enable Docker inside
+    the workspace.
+- [AWS EC2 dev container](https://github.com/coder/coder/tree/main/examples/templates/aws-devcontainer)
+  - Runs a development container inside a single EC2 instance. It also mounts
+    the Docker socket from the VM inside the container to enable Docker inside
+    the workspace.
+
+Your template can prompt the user for a repo URL with
+[parameters](../../extending-templates/parameters.md):
```

Comment:
May also mention to use the [`git-clone`](https://registry.coder.com/modules/git-clone) module from the registry.